articles/150_ros_command_line_arguments.md

```diff
@@ -201,6 +202,25 @@ ros2 run some_package some_ros_executable --ros-args --disable-rosout-logs --dis
 
 Logging is fully enabled by default, thus `--enable-*` options are usually redundant unless a `--disable-*` option found earlier in the command line is being overridden.
 
+#### Enclave assignments
+
+Enclave assignment may be achieved using the `--enclave`/`-e` option.
+This option takes a single string `value` assignment statement, where `value` is where value is fully qualified enclave path used to locate the respective security artifacts within the configured keystore.
+
+As an example, to assign a enclave path `/foo/bar` one may execute:
+
+```sh
+ros2 run some_package some_ros_executable --ros-args --enclave="/foo/bar"
+```
+
+or its shorter equivalent:
+
+```sh
+ros2 run some_package some_ros_executable --ros-args -e "/foo/bar"
+```
+
+As is, this enclave assignment applies to each and every context that `some_ros_executable` spawns unless explicitly ignored in code or overridden via security environment variables.  
```

Comment:
> overridden via security environment variables

Shouldn't the flag take precedence over the environment variables?

Reply:
the motivation for this to be the other way around is to be able to use this flag via e.g. launch to specify enclave path for a specific node.

The environment variable will allow to override that flag for all the processes in the launchfile. So the env var has to be gien precedence otherwise it wouldnt be possible to override from the outside

Reply:
> The environment variable will allow to override that flag for all the processes in the launchfile.

Should it really? If someone bothers to write flags down in a launchfile maybe we shouldn't override them in such 'hidden' manner. I'm not saying that it shouldn't be possible to override them, but that it should me more 'visible' (as in pro-active).

Given that in ROS it is usually encouraged to export env vars in `.bashrc` ([even in sros2 tuto](https://github.com/ros2/sros2/blob/master/SROS2_Linux.md#define-the-sros2-environment-variables)), I foresee a lot of frustration with this priority being the other way around, names will be called.

Reply:
The flags are normally controlled by the maintainer of the launchfile or scripts, that may be from an upstream package, the users are the ones normally in control of the environment those upstream launchfiles are started from. The override env allows the user the master control of what enclave should be used for the scope of processes spawned from that environment.

Take for example you are creating security enclaves for a swarm of robots. A this scale, you only care about access control at the resolution of the robot, not necessarily at the individual node level, given the threat model you'd like to satisfy with a compromised robot being a whole device. Instead of having to update all the upstream launch files on the robot to use the one enclave allocated for that robot, you could instead use the override env.

There are also a number of other development workflows that benefit from this environmental control knob, like when debugging security issues, switching between enclaves with varying degrees of access control to triage what permissions may be missing from a profile.